src/chain/node.js

```diff
@@ -141,18 +144,24 @@ async function topBlock () {
   return this.api.getTopHeader()
 }
 
-async function poll (th, { blocks = 10, interval = 500, allowUnsynced = false } = {}) {
+async function poll (th, { blocks = 10, interval = 500 } = {}) {
   const max = await this.height() + blocks
   do {
-    const tx = await this.tx(th).catch(_ => null)
-    if (tx && (tx.blockHeight !== -1 || (allowUnsynced && tx.height))) {
-      return tx
-    }
+    const tx = await this.api.getTransactionByHash(th)
+    if (tx?.blockHeight !== -1) return tx
     await pause(interval)
   } while (await this.height() < max)
-  throw new Error(`Giving up after ${blocks} blocks mined, transaction hash: ${th}`)
+  const status = this.api.getCheckTxInPool && (await this.api.getCheckTxInPool(th)).status
+  throw new Error([
+    `Giving up after ${blocks} blocks mined`,
+    `transaction hash: ${th}`,
+    ...status ? [`status: ${status}`] : []
+  ].join(', '))
 }
 
+/**
+ * @deprecated use `sdk.api.getTransactionInfoByHash` instead
+ */
 async function getTxInfo (hash) {
   const result = await this.api.getTransactionInfoByHash(hash)
   return result.callInfo || result
```

Comment:
`getTransactionInfoByHash` also has `gaInfo` and `txInfo` fields, merging this would make it overcomplicated, I'm proposing to use the plain endpoint of node instead.